examples/pub-sub-connection-state/react/src/app/page.tsx

```diff
@@ -0,0 +1,84 @@
+'use client'
+
+import { useState } from 'react';
+import { Realtime } from 'ably';
+import { AblyProvider } from 'ably/react';
+import { nanoid } from 'nanoid';
+
+export default function Home() {
+  const [client, setClient] = useState<Realtime | null>(null);
+  const [connectionState, setConnectionState] = useState('Disconnected');
```

Comment:
We have built-in `useConnectionStateListener` in react-hooks, probably it's better to use it here